src/pykiso/test_coordinator/test_execution.py

```diff
@@ -128,6 +132,53 @@ def set_skipped(test_case: BasicTest) -> None:
     list(map(set_skipped, filter(is_skip_test, base_suite)))
 
 
+def apply_test_case_filter(
+    all_tests_to_run: unittest.TestSuite,
+    test_class_pattern: str,
+    test_case_pattern: str,
+) -> unittest.TestSuite:
+    """Apply a filter to run only test cases which matches given expression
+
+    :param all_tests_to_run: a dict containing all testsuites and testcases
+    :param test_class_pattern: pattern to select test class as unix filename pattern
+    :param test_case_pattern: pattern to select test case as unix filename pattern
+    :return: new test suite with filtered test cases
+    """
+
+    base_suite = test_suite.flatten(all_tests_to_run)
+
+    def is_active_test(
+        test_case: BasicTest, test_class_pattern: str, test_case_pattern: str
+    ) -> bool:
+        """Check if testcase shall be active by given selection patterns
+
+        :param test_case: unittest test
+        :param test_class_pattern: pattern to select test class as unix filename pattern
+        :param test_case_pattern: pattern to select test case as unix filename pattern
+        :return: True if test matches patterns else False
+        """
+        test_class_name = re.sub(r"-\d+-\d+", "", test_case.__class__.__name__)
+        is_classname_match = bool(fnmatch(test_class_name, test_class_pattern))
+
+        if is_classname_match and test_case_pattern is None:
+            return is_classname_match
+
+        elif is_classname_match and test_case_pattern:
+            return bool(fnmatch(test_case._testMethodName, test_case_pattern))
+        else:
+            return False
+
+    filtered_suite = filter(
+        partial(
```

Comment:
Instead of setting the function params with a partial, you could also remove them entirely and pass then from the "upper-level" function:

```
def apply_test_case_filter(
    all_tests_to_run: unittest.TestSuite,
    test_class_pattern: str,
    test_case_pattern: str,
) -> unittest.TestSuite:
    base_suite = test_suite.flatten(all_tests_to_run)
    test_class_pattern = test_class_pattern    # don't know if setting the variable explicitly is necessary, TBD.. :)
    test_case_pattern = test_case_pattern 

    def is_active_test(test_case: BasicTest) -> bool:
        test_class_name = re.sub(r"-\d+-\d+", "", test_case.__class__.__name__)
        is_classname_match = bool(fnmatch(test_class_name, test_class_pattern))
        ...
```

Reply:
yup thats true. Removed unnecessary parts.